MekHQ/src/mekhq/campaign/finances/enums/FinancialTerm.java

```diff
@@ -68,33 +70,94 @@ public boolean isQuarterly() {
         return this == QUARTERLY;
     }
 
+    public boolean isSemiannually() {
+        return this == SEMIANNUALLY;
+    }
+
     public boolean isAnnually() {
         return this == ANNUALLY;
     }
     //endregion Boolean Comparison Methods
 
     /**
-     * @param origin the origin date, which will normally (but not always) be the current date
+     * @param today the origin date, which will normally (but not always) be the current date
      * @return the next valid date for the financial term, with a built-in grace period to line up
      * everything to the first day of a financial setup (Monday, First of the Month, First of the Year)
      */
-    public LocalDate nextValidDate(final LocalDate origin) {
+    public LocalDate nextValidDate(final LocalDate today) {
+        return nextValidDate(today.minusDays(1), today);
+    }
+
+    /**
+     * @param yesterday the day before the today
+     * @param today the origin date, which will normally (but not always) be the current date
+     * @return the next valid date for the financial term, with a built-in grace period to line up
+     * everything to the first day of a financial setup (Monday, First of the Month, First of the Year)
+     */
+    public LocalDate nextValidDate(final LocalDate yesterday, final LocalDate today) {
         switch (this) {
             case BIWEEKLY:
-                return origin.with(TemporalAdjusters.nextOrSame(DayOfWeek.MONDAY)).plusWeeks(2);
+                // First, find the first day of the current week
+                final LocalDate date = today.with(WeekFields.ISO.dayOfWeek(), 1);
+                // Second, add two weeks to the date if this is an even week, or three if it is not
+                return date.plusWeeks(
+                        ((date.get(WeekFields.ISO.weekOfYear()) % 2) == 0)
+                                ? 2 : 3);
```

Comment:
BiWeekly is every 2 weeks. BiMonthly is twice a month (or every 2 months).

So shouldn't this just be every 2 weeks?

Reply:
We use a built-in grace period that lines everything up to the first day of the financial setup (Monday, First of the Month, First of the Year, etc). I didn't want to change the internal decision for that, so this actually handles that decision properly.

Reply:
So this is basically saying "if this is in the middle of a bi-weekly period, add a week" and locks the biweekly literally into every even week vs time based every two weeks?

Reply:
Exactly, same way as it was meant to be handled previously but without the calculation bugs.